rust/gui-client/src-tauri/src/client.rs

```diff
@@ -107,17 +111,54 @@ pub(crate) fn run() -> Result<()> {
 ///
 /// Automatically logs or shows error dialogs for important user-actionable errors
 fn run_gui(cli: Cli) -> Result<()> {
-    let result = gui::run(cli);
+    let (settings, _logger) = start_logging()?;
+    let result = gui::run(cli, settings);
 
     // Make sure errors get logged, at least to stderr
     if let Err(error) = &result {
-        tracing::error!(?error, "gui::run error");
+        tracing::error!(?error, error_msg = error.to_string(), "gui::run error");
         show_error_dialog(error)?;
     }
 
     Ok(result?)
 }
 
+/// Loads the settings and starts logging
+///
+/// Don't drop the log handle or logging will stop.
+fn start_logging() -> Result<(AdvancedSettings, file_logger::Handle)> {
```

Comment:
This all needed to move up to `client.rs` so that we can log errors from `gui::run` and also log whatever error message we show in the error dialogs. If it's in `gui.rs` the logger will get dropped. 

My previous hypothesis about a bug in `tracing-stackdriver` was incorrect.